custom_components/elasticsearch/entity_details.py

```diff
@@ -15,8 +20,13 @@ class FullEntityDetails:
 
     entity: entity_registry.RegistryEntry
     entity_area: area_registry.AreaEntry | None
+    entity_label: label_registry.LabelEntry | None
+
     device: device_registry.DeviceEntry | None
     device_area: area_registry.AreaEntry | None
+    device_floor: floor_registry.FloorEntry | None
+    device_label: label_registry.LabelEntry | None
```

Comment:
As noted below, this looks incorrect as a device may have multiple labels

```suggestion
    device_labels: list[label_registry.LabelEntry] | None
```